### DISCUSSION TITLE
Cart / Checkout navigation errors ootb

### DISCUSSION THREAD

**Discussion by jacksinn** [vuestorefront/shopify#239] [Q&A] [answered]
There are three errors we are noticing out-of-the-box:
1. Initial add-to-cart  not adding product to cart
2. Going to checkout then clicking the back button / going back in history takes the app to two screens prior to going to checkout 
3. Repeating the checkout from step 2 navigates back again and the cart is empty (even on refresh)

Video https://youtu.be/jZqiUgUfnLk

 Steps taken (with screens):

`yarn; yarn dev` fresh ootb install with .env set to Shopify store
Click through app using following:
Navigate to home
Navigate to products (PLP)
Click a product (go to PDP)
Add product to bag
A notification shows that the product has been added to the bag but the cart is empty

<img width="1895" alt="EYES   LIPS" src="https://user-images.githubusercontent.com/166227/156067837-22b866d3-2648-4af9-af6a-54a74eac7bba.png">

<img width="1872" alt="Pasted Graphic 1" src="https://user-images.githubusercontent.com/166227/156067857-073e116b-b26d-446d-acf1-85b4cd6b1132.png">

Do a hard refresh on the PDP then click add to cart.
Cart populates with product.

<img width="1847" alt="EYES   LIPS" src="https://user-images.githubusercontent.com/166227/156067896-c9ffa236-8540-4c69-a6a2-ea257a834138.png">

<img width="1883" alt="GLOBAL-REPAIR" src="https://user-images.githubusercontent.com/166227/156067921-7828c4d8-5a2a-4929-978d-443410655f4a.png">

——
Click “Go to checkout” from cart.
(Goes to Shopify checkout)

<img width="1548" alt="Pasted Graphic 4" src="https://user-images.githubusercontent.com/166227/156067948-a3b55325-6380-4d2b-8a5a-ea97608ca1cb.png">

Click back button from checkout.
Pages returns to n-2 in the route history (goes to PLP instead of PDP e.g. which we had just navigated from to checkout)

<img width="1786" alt="Pasted Graphic 5" src="https://user-images.githubusercontent.com/166227/156067955-504578c8-6383-4945-915f-1eb8c3245439.png">

Repeat the steps of clicking on the cart then clicking to checkout.
Navigates to Shopify checkout.
Click back button. 
Navigates to n-2, now the home page e.g.
Page loads, cart is empty.
Refreshing does not restore cart contents.

<img width="1765" alt="COCKTAIL" src="https://user-images.githubusercontent.com/166227/156067970-1cb064d2-7b0a-4e22-87a6-5287f5c3410a.png">

**Comment by aureate-labs-team** [comment, marked as answer]
It's because of multiple calls with the loadCart() method. We've fixed it. You can fix it by manually migrate the changes to your existing setup or setup a new fresh instance using CLI :)
